scripts/o3de/tests/test_project_manager_interface.py

```diff
@@ -0,0 +1,413 @@
+    sig = signature(manifest.get_engine_gems)
+
+    assert sig.return_annotation == list
+
+def test_get_all_gems():
+    sig = signature(manifest.get_all_gems)
+    assert sig.parameters.__len__() >= 1
+
+    project_path = list(sig.parameters.values())[0]
+    assert project_path.name == 'project_path'
+    assert project_path.annotation == pathlib.Path
+
+    assert sig.return_annotation == list
+
+def test_get_gem_json_data():
+    sig = signature(manifest.get_gem_json_data)
+    assert sig.parameters.__len__() >= 3
+
+    parameters = list(sig.parameters.values())
+    gem_path = parameters[1]
+    assert gem_path.name == 'gem_path'
+    assert gem_path.annotation == str or gem_path.annotation == pathlib.Path
+    project_path = parameters[2]
+    assert project_path.name == 'project_path'
+    assert project_path.annotation == str or project_path.annotation == pathlib.Path
+
+    assert sig.return_annotation == dict
+
+def test_get_project_json_data():
+    sig = signature(manifest.get_project_json_data)
+    assert sig.parameters.__len__() >= 2
+
+    project_path = list(sig.parameters.values())[1]
+    assert project_path.name == 'project_path'
+    assert project_path.annotation == str or project_path.annotation == pathlib.Path
+
+    assert sig.return_annotation == dict
+
+def test_get_manifest_projects():
+    sig = signature(manifest.get_manifest_projects)
+
+    assert sig.return_annotation == list
+
+def test_get_engine_projects():
+    sig = signature(manifest.get_engine_projects)
+
+    assert sig.return_annotation == list
+
+def test_get_template_json_data():
+    sig = signature(manifest.get_template_json_data)
+    assert sig.parameters.__len__() >= 3
+
+    parameters = list(sig.parameters.values())
+    template_path = parameters[1]
+    assert template_path.name == 'template_path'
+    assert template_path.annotation == str or template_path.annotation == pathlib.Path
+    project_path = parameters[2]
+    assert project_path.name == 'project_path'
+    assert project_path.annotation == str or project_path.annotation == pathlib.Path 
+
+    assert sig.return_annotation == dict
+
+def test_get_templates_for_project_creation():
+    sig = signature(manifest.get_templates_for_project_creation)
+
+    assert sig.return_annotation == list
+
+def test_get_repo_json_data():
+    # Interface temporarily removed
+    return
+
+    sig = signature(manifest.get_repo_json_data)
+    assert sig.parameters.__len__() >= 1
+
+    repo_uri = list(sig.parameters.values())[0]
+    assert repo_uri.name == 'repo_uri'
+    assert repo_uri.annotation == str
+
+    assert sig.return_annotation == dict
+
+def test_get_repo_path():
+    # Interface temporarily removed
+    return
+
+    sig = signature(manifest.get_repo_path)
+    assert sig.parameters.__len__() >= 1
+
+    repo_uri = list(sig.parameters.values())[0]
+    assert repo_uri.name == 'repo_uri'
+    assert repo_uri.annotation == str
+
+    assert sig.return_annotation == pathlib.Path
```

Comment:
maybe remove these until the interface is sorted out?

Reply:
If this PR gets merged then I'll re-enable the tests: https://github.com/o3de/o3de/pull/7539
It adds those methods back in.

Otherwise if the current PR is merged first former PR can enable it.

Reply:
#7563 has been merged!

Reply:
Tests enabled!